### PR TITLE
Adding System.Data.Odbc reference source files

### PR DESCRIPTION
**Pull request by tinchou**
This PR adds the reference source for System.Data.Odbc in preparation for adding the new System.Data.Odbc library. I don't know if this is how you usually approach creating a new library, so let me know.

Checking this PR is a no-op, since the files were just copied from https://github.com/Microsoft/referencesource and pasted where they will live in the new library folder.

Also, having these in place should really help for diffing my changes in upcoming PRs.

@divega @anpete @eerhardt @saurabh500 you might be interested.

### CODE REVIEW

**Comment by saurabh500** [comment]
You should change the headers in the source files to have the MIT license

**Comment by tinchou** [comment]
should I do that in this PR or the next one?

**Comment by stephentoub** [comment]
> should I do that in this PR or the next one?

Please do it in this one.

**Comment by tinchou** [comment]
Done

**Comment by tinchou** [comment]
We can do a merge squash from the PR UI, right?

**Comment by saurabh500** [comment]
> We can do a merge squash from the PR UI, right?
I don't think so.

I squash the commits on the branch on my fork and push it to the remote of the fork. That automatically updates the PR with the updated branch where only one commit would show.

**Comment by anpete** [comment]
@tinchou Easy to do locally with something like `git rebase -i HEAD~2`

**Comment by tinchou** [comment]
I already squashed them as suggested, but I guess it's here and it could be used (less manual work, less CI cycles, etc.): https://github.com/blog/2141-squash-your-commits

**Comment by danmosemsft** [comment]
Most times folks here just hit "Squash and merge" button. I never squash locally.

**Comment by anpete** [comment]
@tinchou Did you see the [coding guidelines](https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/Documentation/coding-guidelines/coding-style.md)? Would be good to align to these early on.

**Comment by justinvp** [comment]
It'd be nice if the [CodeFormatter](https://github.com/dotnet/codeformatter) was run over these before merging.

File: src/System.Data.Odbc/src/System/Data/Odbc/DbDataRecord.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;            //Component
+using System.Data;
+using System.Runtime.InteropServices;   //Marshal
+using System.Reflection;                //Missing
+
+namespace System.Data.Odbc {
+    sealed internal class DbSchemaInfo {
```
Comment (justinvp) [inline src/System.Data.Odbc/src/System/Data/Odbc/DbDataRecord.cs]: `internal sealed`

File: src/System.Data.Odbc/src/System/Data/Odbc/Odbc32.cs [hunk 2]
```diff
@@ -0,0 +1,997 @@
...
+namespace System.Data.Odbc {
+
+    using System;
+    using System.Data;
+    using System.Data.Common;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Runtime.InteropServices;
+    using System.Runtime.ConstrainedExecution;
+    using System.Text;
```
Comment (justinvp) [inline src/System.Data.Odbc/src/System/Data/Odbc/Odbc32.cs]: Namespace imports should be specified at the top of the file, outside of namespace declarations and should be sorted alphabetically.

File: src/System.Data.Odbc/src/System/Data/Odbc/odbcmetadatacollectionnames.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (justinvp) [inline src/System.Data.Odbc/src/System/Data/Odbc/odbcmetadatacollectionnames.cs]: It'd be nice if the file name was cased correctly, e.g. OdbcMetaDataCollectionNames.cs

File: src/System.Data.Odbc/src/System/Data/Odbc/odbcmetadatacolumnnames.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (justinvp) [inline src/System.Data.Odbc/src/System/Data/Odbc/odbcmetadatacolumnnames.cs]: It'd be nice if the file name was cased correctly, e.g. OdbcMetaDataColumnNames.cs

File: src/System.Data.Odbc/src/System/Data/Odbc/odbcmetadatafactory.cs [hunk 5]
```diff
@@ -0,0 +1,1075 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (justinvp) [inline src/System.Data.Odbc/src/System/Data/Odbc/odbcmetadatafactory.cs]: It'd be nice if the file name was cased correctly, e.g. OdbcMetaDataFactory.cs

**Comment by tinchou** [comment]
@justinvp thanks a lot! I believe I applied all your suggestions :)

**Comment by stephentoub** [comment]
As a code dump, LGTM. Thanks for getting this started.

**Comment by tinchou** [comment]
First merged PR wahoooo

**Pull request merged**